wrf_ens_tools/post/post_process.py

```diff
@@ -1243,49 +1239,60 @@ def open_wrf_dataset(inname, nest='static', dask=True, chunks=None):
                                                                   '%Y-%m-%d_%H:%M:%S').strftime('%Y-%m-%dT%H:%M:%S')
         elif attr == 'MOAD_CEN_LAT':
             ds.attrs['central_latitude'] = indata.MOAD_CEN_LAT
-        # elif attr == 'CEN_LON':
-        #     ds.attrs['central_longitude'] = indata.CEN_LON
+        elif attr == 'CEN_LON':
+            continue
+        elif attr == 'CEN_LAT':
+            continue
         elif attr == 'TRUELAT1':
             ds.attrs['true_latitude_1'] = indata.TRUELAT1
         elif attr == 'TRUELAT2':
             ds.attrs['true_latitude_2'] = indata.TRUELAT2
         elif attr == 'STAND_LON':
             ds.attrs['standard_longitude'] = indata.STAND_LON
         elif attr == 'MAP_PROJ_CHAR':
-            ds.attrs['projection'] = indata.MAP_PROJ_CHAR
-        elif attr == 'WEST-EAST_GRID_DIMENSION':
+            continue  # projection metadata added during grid computation
+        elif attr == 'MAP_PROJ':  # Ignore map projection numeric ID
             continue
-        elif attr =='SOUTH-NORTH_GRID_DIMENSION':
+        elif 'WEST-EAST' in attr:
             continue
-        elif attr == 'BOTTOM-TOP_GRID_DIMENSION':
+        elif 'SOUTH-NORTH' in attr:
+            continue
+        elif 'BOTTOM-TOP' in attr:
             continue
         elif attr == 'DX':
             ds.attrs['dx'] = indata.DX
         elif attr == 'DY':
             ds.attrs['dy'] = indata.DY
+        elif attr == 'POLE_LAT':
+            ds.attrs['pole_latitude'] = indata.POLE_LAT
+        elif attr == 'POLE_LON':
+            ds.attrs['pole_longitude'] = indata.POLE_LON
         else:
             ds.attrs[attr] = indata.attrs[attr]
 
     # Calculate the model projection x and y coordinates
     # TODO: Add support for more projections
-    r = 6370000
-    x_model, y_model = lcc_projection(indata, r=r)
-    ds['x'] = x_model
-    ds.x.attrs['description'] = 'Projection x coordinate'
-    ds['y'] = y_model
-    ds.y.attrs['description'] = 'Projection y coordinate'
-
-    # add projection information
-    ds.attrs['projection'] = 'Lambert Conformal Conic'
-    ds.attrs['semimajor_axis'] = r
-    ds.attrs['semiminor_axis'] = r
-    ds.attrs['ellipse'] = 'sphere'
+    if indata.MAP_PROJ_CHAR == 'Lambert Conformal':
+        r = 6370000
+        x_model, y_model = lcc_projection(indata, r=r)
+        ds['x'] = x_model
+        ds.x.attrs['description'] = 'Projection x coordinate'
+        ds['y'] = y_model
+        ds.y.attrs['description'] = 'Projection y coordinate'
+
+        # add projection information
+        ds.attrs['projection'] = 'Lambert Conformal Conic'
+        ds.attrs['semimajor_axis'] = r
+        ds.attrs['semiminor_axis'] = r
+        ds.attrs['ellipse'] = 'sphere'
+    else:
+        raise NotImplementedError('Input projection {} not supported'.format(indata.MAP_PROJ_CHAR))
```

Comment:
This may not be important right now, but in the future we may want to establish use of a standard for defining the semi-major and semi-minor axes, such as the WSG84.

Reply:
That would be good if we were including coordinate transforms, right now I am just calculating the x and y coordinates for the WRF projection (which are not included in the raw output) and directly copying the other coordinate information. The user is left to transform as desired.